mesonbuild/backend/ninjabackend.py

```diff
@@ -1255,12 +1271,13 @@ def generate_vala_compile(self, target):
                 extra_args.append(relname)
             else:
                 extra_args.append(a)
+        internal_dependency_vapis = self.determine_internal_dep_vapis(target)
         dependency_vapis = self.determine_dep_vapis(target)
-        extra_dep_files += dependency_vapis
+        extra_dep_files += dependency_vapis + internal_dependency_vapis
         args += extra_args
         element = NinjaBuildElement(self.all_outputs, valac_outputs,
                                     self.compiler_to_rule_name(valac),
-                                    all_files + dependency_vapis)
+                                    all_files + internal_dependency_vapis)
```

Comment:
should this be `extra_dep_files`?

Reply:
No because `dependency_vapis` are system files and not built by ninja